### ISSUE TITLE
SpinButton: Convert to new CSS-in-JS pattern

### ISSUE THREAD

**Issue by mikewheaton** [microsoft/fluentui#6808]
**Describe the feature that you would like added**
SpinButton currently uses the `@customizable` decorator, instead of the new `styled` function with a `SpinButton.base.tsx` file. It needs to be converted to the new pattern so that it can be overridden with the Fluent theme in #6807.

**What component or utility would this be added to**
SpinButton

**Comment by Vitalius1** [comment]
Adding to the pile of conversion some other issues I noticed when I fuentized the component in #6890:

This one is odd too, as it's also present in the default styles and decided not to risk breaking an edge case this calc was possibly introduced in the first place.

Personally, I consider the default styles badly need some refactoring for a couple reasons:

1. There almost no className hooks for Sass overrides. Here how the tree looks now:
<img width="437" alt="screen shot 2018-10-30 at 8 50 10 am" src="https://user-images.githubusercontent.com/29514833/47731331-1ccc9080-dc21-11e8-9a6e-e25c2ad6092f.png">

2. The styles are trying to support an older implementation with getClassNames prop by mixing the logic with the styles prop and not offering any styleProps to conditionally alter the styles (just a bunch of different classNames for all the possible states).

3. The calc function is most likely needed because the positioning of some elements (specifically the input field with the up and down buttons) is using float. Would be nice to have it converted to flex so it does it for you.

4. There is some misalignment of the label being slightly uncentered relative to the input field text, probably caused by floats again.

<img width="309" alt="screen shot 2018-10-30 at 9 00 36 am" src="https://user-images.githubusercontent.com/29514833/47731998-8ef1a500-dc22-11e8-8241-724b0cbe14e6.png">

**Comment by mikewheaton** [comment]
@Vitalius1 Do you want to open a new issue for that? This original issue was opened because I misunderstood the old `@customizable` pattern and thought that we couldn't apply style overrides with the Fluent theme. We can, so there's no need to update it for this project.

**Comment by Vitalius1** [comment]
@mikewheaton I was able to override with fluent theme here because of the way it has the styles are applied in here. But I was still not able to use the theme and it's palette because the styles prop is not a style function or object as for the most components. So I think the issue still needs to be open and the info I gave is just an bonus upgrade that needs to be done. If you feel that this issue still needs to be closed I will gladly open a separate issue but I consider the things I am mentioned should be done at the time of the conversion to the new pattern.
